Sources/_MatchingEngine/Regex/Parse/Parse.swift

```diff
@@ -116,10 +116,27 @@ extension Parser {
 extension Parser {
   /// Parse a regular expression
   ///
-  ///     Regex        -> '' | Alternation
-  ///     Alternation  -> Concatenation ('|' Concatenation)*
+  ///     Regex        -> GlobalMatchingOption* RecursiveRegex
```

Comment:
I think I slightly prefer this, but let me know what you think:

```
/// RegexLiteral -> GlobalOption* Regex
/// Regex -> ...
```

Reply:
Yeah that crossed my mind, but I felt that `RegexLiteral` sounded like it would cover the grammar including the delimiters. With the implementation of `AST.Node` perhaps instead `RecursiveRegex` could be `RegexNode`? I don't feel strongly about it though, and am fine with `RegexLiteral` & `Regex`.

Reply:
`Regex` and `RegexComponent` or `RegexNode` sound good to me.